### DISCUSSION TITLE
Custom paramTypes precedence

### DISCUSSION THREAD

**Discussion by ivaskevych** [sql-formatter-org/sql-formatter#781] [Q&A]
Hi, when I was trying to configure custom `paramTypes` support like `:END_DATE:` for `PL\SQL` dialect faced with issue that default `PL\SQL` named params configuration is taking precedence over custom regex that is resulting into an error.

![image](https://github.com/user-attachments/assets/8bcbe602-0668-4d86-8290-b06e9d237db2)

Just curious if there is a way to support both default and custom params other than overriding default with custom like:
```
{
  language: 'plsql',
  paramTypes: {
    custom: [
      {
        regex: ':[A-Z_]+:',
      },
      // named param override
      {
        regex: ':[A-Za-z_]+',
      },
    ],
    named: [],
}
```

**Comment by nene** [comment]
Currently no. The parser tries to match the parameters in order of:

- named
- quoted
- numbered
- positional
- custom

**Reply by ivaskevych** [reply]
Got it, thanks for quick reply
